src/tqdm_publisher/_demo/_server.py

```diff
@@ -0,0 +1,89 @@
+import asyncio
+import json
+import threading
+import time
+from typing import Any, Dict, List
+from uuid import uuid4
+
+import websockets
+
+from tqdm_publisher import TQDMPublisher
+
+
+def start_progress_bar(*, client_id: str, progress_bar_id: str, client_callback: callable) -> None:
+    """
+    Emulate running the specified number of tasks by sleeping the specified amount of time on each iteration.
+
+    Defaults are chosen for a deterministic and regular update period of one second for a total time of one minute.
+    """
+    all_task_durations_in_seconds = [1.0 for _ in range(60)]  # One minute at one second per update
+    progress_bar = TQDMPublisher(iterable=all_task_durations_in_seconds)
+
+    def run_function_on_progress_update(format_dict: dict) -> None:
+        """
+        This is the injected callback that will be run on each update of the TQDM object.
+
+        Its first and only positional argument must be the `format_dict` of the TQDM instance. Additional customization
+        on outside parameters must be achieved by defining those fields at an outer scope and defining this
+        server-specific callback inside the local scope.
+
+        In this demo, we will execute the `client_callback` whose protocol is known only to the WebSocketHandler.
+        It has
+        """
+        client_callback(client_id=client_id, progress_bar_id=progress_bar_id, format_dict=format_dict)
+
+    progress_bar.subscribe(callback=run_function_on_progress_update)
+
+    for task_duration in progress_bar:
+        time.sleep(task_duration)
+
+
+class WebSocketHandler:
+    """Describe this class."""
+
+    def __init__(self) -> None:
+        """Initialize the mapping of client IDs to ."""
+        self.clients: Dict[str, Any] = dict()
+
+    def function_to_run_on_progress_update(self, *, client_id: str, progress_bar_id: str, format_dict: dict) -> None:
+        """This is..."""
+        asyncio.run(self.send(client_id=client_id, data=dict(progress_bar_id=progress_bar_id, format_dict=format_dict)))
+
+    async def send(self, client_id: str, data: dict) -> None:
+        """Send an arbitrary JSON object `data` to the client identifier by `client_id`."""
+        await self.clients[client_id].send(json.dumps(obj=data))
+
+    async def handler(self, websocket) -> None:
```

Comment:
```suggestion
    async def handler(self, websocket) -> None:
```

Does the `websocket` have a type we can use for annotation?

Reply:
Yes, this is a [WebSocketServerProtocol](https://websockets.readthedocs.io/en/stable/reference/asyncio/server.html#websockets.server.WebSocketServerProtocol)

Reply:
Thanks for checking and updating that everywhere! 😁